packages/astro/src/runtime/client/dev-overlay/plugins/xray.ts

```diff
@@ -0,0 +1,125 @@
+
+export default {
+	id: 'astro:xray',
+	name: 'Xray',
+	icon: icon,
+	init(canvas) {
+		let islandsOverlays: { highlightElement: DevOverlayHighlight; island: HTMLElement }[] = [];
+		addIslandsOverlay();
+
+		function addIslandsOverlay() {
+			const islands = document.querySelectorAll<HTMLElement>('astro-island');
+
+			islands.forEach((island) => {
+				const computedStyle = window.getComputedStyle(island);
+				const islandElement = (island.children[0] as HTMLElement) || island;
+
+				// If the island is hidden, don't show an overlay on it
+				if (islandElement.offsetParent === null || computedStyle.display === 'none') {
+					return;
+				}
```

Comment:
Is this a sufficient visibility check?

Reply:
It's not! I'm not sure if there's a proper way to do this, I'll investigate more outside of this PR.